### PR TITLE
Add SQL threat detection policies resource

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

File: resources/sql_databases.go [hunk 1]
```diff
@@ -406,6 +406,94 @@ func SQLDatabases() *schema.Table {
...
 			},
+			{
+				Name:        "azure_sql_database_db_threat_detection_policies",
+				Description: "Contains information about a database Threat Detection policy.",
+				Resolver:    fetchSqlDatabaseDbThreatDetectionPolicies,
+				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"database_cq_id", "id"}},
+				Columns: []schema.Column{
+					{
+						Name:        "database_cq_id",
+						Description: "Unique ID of azure_sql_databases table (FK)",
```
Comment (roneli) [inline resources/sql_databases.go:417]: "Unique ID of azure_sql_databases table (FK)" -> "Unique CloudQuery ID of azure_sql_databases table (FK)"

File: resources/sql_databases.go [hunk 2]
```diff
@@ -453,3 +541,19 @@ func fetchSqlDatabaseDbBlobAuditingPolicies(ctx context.Context, meta schema.Cli
...
 }
+
+func fetchSqlDatabaseDbThreatDetectionPolicies(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	svc := meta.(*client.Client).Services().SQL.DatabaseThreatDetectionPolicies
+	database := parent.Item.(sql.Database)
+	details, err := client.ParseResourceID(*database.ID)
+	if err != nil {
+		return err
+	}
+	serverName := parent.Parent.Get("name").(*string)
```
Comment (roneli) [inline resources/sql_databases.go:552]: why not use parent.Parent.Item.(*...) also parent.Parent.Get can return null, so its important to check that.

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, minor comments on code

File: resources/sql_databases.go (hunk 1 above)
Comment (irmatov) [reply resources/sql_databases.go:417]: Fixed

File: resources/sql_databases.go (hunk 2 above)
Comment (irmatov) [reply resources/sql_databases.go:552]: Changed to use parent.Parent.Item as suggested, thanks

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**